DragonSlayer.cpp

```diff
@@ -27,17 +27,17 @@ void DragonSlayer::attack(Character& other)
         
          for( auto& item : attackItems )
         {
-            if( auto* attackItem1 = dynamic_cast<AttackItem*>(item.get()) )
+            if( auto* attackItem = dynamic_cast<AttackItem*>(item.get()) )
             {
-                attackItem1->use(this);
+                attackItem->use(this);
```

Comment:
I'm pretty sure I said that the DragonSlayer only gets one attack item.  He should not have a vector of them.